app/download_test.go

```diff
@@ -0,0 +1,75 @@
+package app
+
+import (
+	"fmt"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/stretchr/testify/require"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestDownloadFromURL(t *testing.T) {
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	app := th.App
+	app.Config().PluginSettings.AllowInsecureDownloadUrl = model.NewBool(true)
+
+	retries := 0
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/succeeds-after-retry", func(w http.ResponseWriter, r *http.Request) {
+		if retries < 2 {
+			http.Error(w, "Request Timed out", http.StatusGatewayTimeout)
+			retries++
+			return
+		}
+
+		_, _ = w.Write([]byte("Your request is successful."))
+	})
+
+	mux.HandleFunc("/fails-forever", func(w http.ResponseWriter, r *http.Request) {
+		http.Error(w, "This would fail forever", http.StatusInternalServerError)
+	})
+
+	testServer := httptest.NewServer(mux)
+
+	type args struct {
+		downloadURL string
+	}
+
+	tests := []struct {
+		name    string
+		args    args
```

Comment:
Is there a reason why we have a separate struct here with just one field ? Can't we directly use the downloadURL here ?